includes/blocks/course-theme/class-lesson-actions.php

```diff
@@ -161,6 +161,21 @@ public function render( array $attributes = [] ): string {
 
 		$course_id = Sensei()->lesson->get_course_id( $lesson_id );
 
+		$is_learning_mode  = \Sensei_Course_Theme_Option::has_learning_mode_enabled( $course_id );
+		$is_awaiting_grade = false;
+
+		$lesson_status = \Sensei_Utils::user_lesson_status( $lesson_id, get_current_user_id() );
+
+		if ( $lesson_status ) {
+			$lesson_status = is_array( $lesson_status ) ? $lesson_status[0] : $lesson_status;
+
+			$is_awaiting_grade = 'ungraded' === $lesson_status->comment_approved;
+		}
+
+		if ( $is_learning_mode && $is_awaiting_grade && 'quiz' === get_post_type() ) {
+			return '';
+		}
```

Comment:
WDYT to extract these lines to a private method returning a boolean? This `render` method seems to be a method growing too much. 🤭